src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -437,6 +491,35 @@ private BuildData MergeBuildManifests(List<BuildData> buildsMetadata)
             return mergedBuild;
         }
 
+        private SigningInformation MergeSigningInfo(List<SigningInformation> signignInformation)
+        {
+            SigningInformation mergedInfo = signignInformation[0];
+
+            for (int i = 1; i < signignInformation.Count; i++)
```

Comment:
Any reason not to foreach for readability?

Reply:
Mainly to avoid logic to check for the first iteration of the loop (line 496) and the rest

Reply:
I just think this can be rewritten to guard against bad callers that didn't explicitly check for an empty signingInformation before calling the method.
Initialize with null, iterate with a foreach. Return ends up being null if there is no info.